### PR TITLE
equicord: 1.10.6 -> 1.10.8; add update script

### PR DESCRIPTION
**Pull request by NotAShelf**
- **equicord: 1.10.6 -> 1.10.8**
- **equicord: add `passthru.updateScript`**

Bumps Equicord and adds an update script.

We intentionally avoid the `tag` attribute, since it caused a build failure for
me due to `finalAttrs.src.tag` missing.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/eq/equicord/package.nix [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
   pname = "equicord";
-  version = "1.10.8"; # from package.json
+  version = "1.10.8";
 
   src = fetchFromGitHub {
     owner = "Equicord";
     repo = "Equicord";
-    rev = "935a5eaf6e5894294ec45ec540e9ecb07e850de0";
-    hash = "sha256-dfOzASBP0dEArJXuddLfPLZ7IcsEfKll+ju1jcHarNk=";
+    rev = "refs/tags/v${finalAttrs.version}";
```
Comment (JohnRTitor) [inline pkgs/by-name/eq/equicord/package.nix:18]: Use `tag = finalAttrs.version` here.

Comment (NotAShelf) [reply pkgs/by-name/eq/equicord/package.nix:18]: As noted in the PR description, using `tag` breaks `env.EQUICORD_HASH = "${finalAttrs.src.tag}"`

I might be missing something, but building with tag will result in:

```bash
error: attribute 'tag' missing
       at /nix/store/00i5cbz0ww2grlv0q9rmyl4s62p95ay1-source/pkgs/by-name/eq/equicord/package.nix:35:24:
           34|     EQUICORD_REMOTE = "${finalAttrs.src.owner}/${finalAttrs.src.repo}";
           35|     EQUICORD_HASH = "${finalAttrs.src.tag}";
             |                        ^
           36|   };
```

Comment (JohnRTitor) [reply pkgs/by-name/eq/equicord/package.nix:18]: Please rebase, tag argument was not exposed previously. Should be fixed with https://github.com/hsjobeki/nixpkgs/commit/ce33e20a03de383dea7d04ec5680d342a8345493

Comment (NotAShelf) [reply pkgs/by-name/eq/equicord/package.nix]: Done.

**Comment by NotAShelf** [comment]
@JohnRTitor think you could merge this? We're lagging behind upstream a little bit.

**Comment by NotAShelf** [comment]
Thank you :)

**Pull request merged**